### PR TITLE
Separate ml code

### PR DESCRIPTION
**Pull request by aasseman**
Splitting the agents, environments and simulation code off to https://github.com/semiotic-ai/autoagora-agents.

Tested successfully in production for equivalence to v0.5.0.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -8,18 +8,14 @@ jobs:
       fail-fast: true
       matrix:
         python-version: ["3.8", "3.9", "3.10"]
-        poetry-version: ["1.2.2"]
```
Comment (tkornuta-semiotic) [inline .github/workflows/tests.yml:11]: automatic versioning, here we come ;)

File: README.md [hunk 2]
```diff
@@ -93,40 +93,3 @@ bandit_mean{subgraph="QmRDGLp6BHwiH9HAE2NYEE3f7LrKuRqziHBv76trT4etgU"} 3.6531261
 ```
 
 Where "bandit" refers to the reinforcement learning method (Continuum-armed bandit) used to track the market price for each subgraph.
```
Comment (tkornuta-semiotic) [inline README.md:95]: hmmm, I understand that you moved the installation instructions somewhere...?

File: pyproject.toml [hunk 3]
```diff
@@ -61,6 +59,7 @@ prometheus-async = {extras = ["aiohttp"], version = "^22.1.0"}
 autoagora-isa = {git = "ssh://git@github.com/semiotic-ai/autoagora-isa.git", rev = "08142a5dcf3cd44c957921d67792d5188d4894ea", optional = true}
 backoff = "^2.2.1"
 python-json-logger = "^2.0.4"
+autoagora-agents = {git = "https://github.com/semiotic-ai/autoagora-agents.git"}
```
Comment (tkornuta-semiotic) [inline pyproject.toml:62]: hmmm, do I remember correctly you wanted to depend on a particular release?
should we build autoagora-agents wheels?

File: tests/test_placeholder.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
+"""Placeholder, remove when tests are implemented.
+
+The goal of this placeholder is to not have pytest exit with code 5 while there is no
+tests implemented. That lets us have the whole CI testing pipeline ready for when tests
+are implemented.
+"""
+
+
+def test_placeholder():
+    pass
```
Comment (tkornuta-semiotic) [inline tests/test_placeholder.py:10]: yeah, CI is clearly missing ;)

**Review by tkornuta-semiotic — DISMISSED** [review]

File: README.md (hunk 2 above)
Comment (aasseman) [reply README.md:95]: That's all indexers need to know: https://github.com/semiotic-ai/autoagora/blob/e64e18b929f66d7cce90043526370bf0511f87db/README.md?plain=1#L13-L19

Comment (aasseman) [reply README.md]: I put some of the "dev" docs back https://github.com/semiotic-ai/autoagora/pull/20/commits/9abef99fc96659080bfbfc7b9f18aeee2ca1ebb8

File: pyproject.toml (hunk 3 above)
Comment (aasseman) [reply pyproject.toml]: Indeed. Did it there: https://github.com/semiotic-ai/autoagora/pull/20/commits/667fcadaec04d16880fd1711df63b9e25b74e0ed

**Review by tkornuta-semiotic — APPROVED** [review]

**Pull request merged**